Common/src/main/java/com/techarium/techarium/client/screen/MachineCoreScreen.java

```diff
@@ -0,0 +1,241 @@
+package com.techarium.techarium.client.screen;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.techarium.techarium.Techarium;
+import com.techarium.techarium.inventory.MachineCoreMenu;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.components.EditBox;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.player.Inventory;
+import org.lwjgl.glfw.GLFW;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineCoreScreen extends AbstractContainerScreen<MachineCoreMenu> {
+
+	private static final ResourceLocation TEXTURE = new ResourceLocation(Techarium.MOD_ID, "textures/gui/machine_core.png");
+	private static final Component SHOW_HINTS_TEXT = Component.translatable("gui.techarium.machine_core.hints");
+
+	private static final int IDS_X = 17;
+	private static final int IDS_Y = 38;
+	private static final int SCROLL_X = 119; // width 6
+	private static final int TEXT_INPUT_X = 19; // width 100
+	private static final int TEXT_INPUT_Y = 26; // height 12
+
+	private final List<ResourceLocation> ids;
+	private final List<MultiBlockButton> buttons;
+	private EditBox textInput;
+	private int startIndex;
+	private boolean showHints;
+
+	public MachineCoreScreen(MachineCoreMenu menu, Inventory inventory, Component title) {
+		super(menu, inventory, title);
+		this.imageWidth = 165;
+		this.imageHeight = 186;
+		this.ids = this.menu.getAllMultiblockStructures();
+		this.buttons = new ArrayList<>();
+	}
+
+	@Override
+	protected void init() {
+		super.init();
+		this.addButtons("");
+		this.textInput = new EditBox(this.font, this.leftPos + TEXT_INPUT_X, this.topPos + TEXT_INPUT_Y, 100, 12, Component.empty());
+		this.addWidget(this.textInput);
+		this.setInitialFocus(this.textInput);
+		this.textInput.setFocus(true);
+		this.textInput.setBordered(false);
+		this.textInput.setResponder(this::filterIds);
+		this.showHints = false;
+	}
+
+	/**
+	 * Add selectable buttons for each multiblock.
+	 *
+	 * @param filter multiblock name filter (do nothing if empty).
+	 */
+	private void addButtons(String filter) {
+		this.startIndex = 0;
+		ResourceLocation selected = this.menu.selectedMultiblock();
+		ArrayList<MultiBlockButton> list = new ArrayList<>();
+		for (int i = 0; i < this.ids.size(); i++) {
+			ResourceLocation id = this.ids.get(i);
+			// name is created from resource location : techarium:random -> multiblock.techarium.random
+			MutableComponent name = Techarium.translatableComponent("multiblock." + id.getNamespace() + "." + id.getPath().replace("/", "."));
```

Comment:
`ResourceLocation.toLanguageKey()` can be used here.

Reply:
I would say you should use the vanilla formatting to keep it consistent with what people expect.

Reply:
but its up to you